src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -186,12 +196,14 @@ public override void VisitLocalFunctionStatement(LocalFunctionStatementSyntax no
                 {
                     addToMap = _enclosing;
                 }
-                addToMap = new InMethodBinder(match, addToMap);
```

Comment:
We need it (we create it later), just don't need it in the map. In fact, it is unexpected to have InMethodBinder in the map for a method body. It is needed to bind the body of the local function, but not to bind the local function statement itself.